atlascope/core/rest/endpoints/dataset_endpoints.py

```diff
@@ -49,3 +61,15 @@ def subimage(self, request, pk):
         subimage.save()
 
         return Response(DatasetSerializer(subimage).data, status=status.HTTP_201_CREATED)
+
+
+    @swagger_auto_schema(
+        responses={200: 'Image file', 404: 'Content not found'},
+    )
+    @action(detail=True, methods=['GET'], renderer_classes=[ContentRenderer])
+    def content(self, request, pk):
+        dataset = self.get_object()
+        image = PIL.Image.open(dataset.content.path)
+        buf = io.BytesIO()
+        image.save(buf, format='PNG')
+        return Response(buf.getvalue())
```

Comment:
What if the image is a large tif?

You could universally use the `thumnail.png` endpoint from django-large-image in #135 

Or if these datasets should have their own endpoints, I'd maybe just use large_image here to produce the thumbnail rather than PIL so that it can handle any Dataset image content

Reply:
Would this work for images that aren't tifs? It seems that if I hit the thumbnail endpoint for a Dataset whose contents are just a small .png image, I get a 404.

The goal of this PR is to be able to display non-tiled images  in geojs on the client (support for tiled images will come later). To that end I've created a new datasaet type, `non_tiled_image`. I could modify this endpoint so that it only works for datasets of that type.

When we were still using MinIO in the stack, I didn't need an endpoint at all, since the `content` field of such dataset objects was already the URL that I needed to get the image, but now that we've switched to local storage, I can't use the `content` field straight up.

Reply:
We would need to add the `large_image_source_pil` dependency (which is lightweight).

> The goal of this PR is to be able to display non-tiled images in geojs on the client (support for tiled images will come later). 

django-large-image is built to handle this! Is the `content` in a `Dataset` always image data?

Reply:
IMO, you should handle *all* image data through `django-large-image` and through tile serving. If you have a simple PNG, then there is only 1 Z layer and one tile.

Here is an example of django-large-image with a simple picture of Afie (the goodest boy there ever was): https://django-large-image.herokuapp.com/image-file/2/viewer

Tiles are served through django-large-image and it is viewed in GeoJS like any image
![Screen Shot 2022-04-15 at 12 30 35 PM](https://user-images.githubusercontent.com/22067021/163607724-343166da-7499-44cf-996a-77a4864eb44d.png)

Reply:
Take a look at 60cca3f and let me know if I'm on the right track. I'm not using `django-large-image` yet (but if that PR moves before this one, I'll make any necessary changes). I agree that doing all image serving from the same place is the best thing to do. Thanks for pointing me in the right direction.

(very handsome dog btw)

Reply:
Those changes look great!